src/lib/css-tag.ts

```diff
@@ -0,0 +1,52 @@
+/**
+@license
+Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+*/
+export const supportsAdoptedStyleSheets = ('adoptedStyleSheets' in Document.prototype);
+
+class LiteralString {
+
+  value: string;
+
+  constructor(value: string) {
+    this.value = value.toString();
+  }
+
+  toString() {
+    return this.value;
+  }
+}
+
+const literalValue = (value: LiteralString) => {
+  if (value instanceof LiteralString) {
+    return value.value;
+  } else {
+    throw new Error(
+        `non-literal value passed to 'css' function: ${value}`
```

Comment:
It'd be nice to allow user-provided values in safe contexts with context-aware auto-escaping. Not sure how far we can go purely client-side. We may be able to allow directive-like functions that return a sentinel value that parses correctly for a particular context, then replace it, lit-html style. Soy does interesting things here at compile time. Maybe @mikesamuel has ideas.